Clean up outdated extensions etc.

emlys opened the pull request:
* Got rid of the `numfig` extension. The same functionality (which we only use in two files) is now available just by setting `numfig = True` in the conf.py. 
* Removed the `json` extension, which as far as I can tell we weren't using anyway
* Got rid of all references to the primer. It's a neat idea, and maybe sometime in the future there will be a use for this, or something like it. If so all the `.. primer` labels are here in dd0ee6c ! But IMO it's not worth keeping that Makefile and conf.py file around, since it was starting to accumulate deprecated things.
* Updated all broken/redirected hyperlinks
* Got rid of MS Word docs and `supplements/`, since they seem to only be there for historical reasons, and will remain available in git history.
* Cleaned up the Makefile and removed `doctest` target, since we have not `.. doctest` blocks anyway.

How does the ReadTheDocs build work? Maybe it does somehow have access to a `json` extension that I don't know about...

emlys commented on source/habitat_quality.rst:
```diff
@@ -1,5 +1,3 @@
-
 .. _habitat_quality:
 
 .. |addbutt| image:: ./shared_images/addbutt.png
```
🤣

dcdenu4 commented:
Hey @emlys can you remind me if this is going to stay as Draft until after the 3.9 release? I kind of remember that from the coffee call. Just wanted to make sure I wasn't ignoring this.

emlys commented:
> Hey @emlys can you remind me if this is going to stay as Draft until after the 3.9 release? I kind of remember that from the coffee call. Just wanted to make sure I wasn't ignoring this.

Yep it's good to wait until after the release to merge! You're welcome to review it now if you want though :)

dcdenu4 commented on source/carbonstorage.rst:
```diff
@@ -178,11 +175,10 @@ Land use/land cover
 
- *  NASA: https://lpdaac.usgs.gov/dataset_discovery/modis/modis_products_table/mcd12q1 (MODIS multi-year global landcover data provided in several classifications)
- *  The European Space Agency: https://www.esa-landcover-cci.org (Three global maps for the 2000, 2005 and 2010 epochs) 
- *  The University of Maryland�s Global Land Cover Facility: http://glcf.umd.edu/data/landcover/ (data available in 1 degree, 8km and 1km resolutions).
```
Is there a reason this link was deleted?

emlys replied:
The site no longer exists, and it seems the whole project was discontinued. I couldn't find the same dataset anywhere.

dcdenu4 commented on source/habitat_risk_assessment.rst:
```diff
@@ -77,28 +75,26 @@ The risk of human activities to habitats or species is modeled in five steps.
+For Euclidean Risk calculation, risk to habitat *j* caused by stressor *k* in each location (i.e. cell) *l* is calculated as the Euclidean distance from the origin in the exposure-consequence space, where average exposure (:eq:`exposure`) is on one axis and the average consequence score (:eq:`consequence`) is on the other.
 
 .. math:: R_{jkl} = \sqrt{(E_{jkl}-1)^2+(C_{jkl}-1)^2}
-   :label: eq3
```
Is there a reason for removing the labels here? I'm just wondering if they might be referenced in the HRA source code? I believe it is something we've done in the past, though that kind of coupling is probably not ideal given the need to upkeep both.

emlys replied:
Having multiple `eq1`s in different model files raises a warning. Each label is supposed to be unique. Giving them more descriptive labels is kinda tedious so I just deleted the ones that aren't referenced anywhere. And the labels that are actually displayed seem to be  different. E.g. the equation labeled `:label: cbc_emissions` in https://storage.googleapis.com/releases.naturalcapitalproject.org/invest-userguide/latest/coastal_blue_carbon.html#the-model shows up with a label (5) [here](https://storage.googleapis.com/releases.naturalcapitalproject.org/invest-userguide/latest/coastal_blue_carbon.html#the-model). I think they always are numbered from the top of the page.

dcdenu4 replied:
Oh, fun! I didn't know that, thanks.

dcdenu4 commented on source/invest_api.rst:
```diff
@@ -17,11 +15,8 @@ Summary
+While there is a user interface for all the InVEST models, we also provide a python application programming interface (API) for more advanced usage of the models.  Full documentation for the ``natcap.invest`` python package is available here: https://invest.readthedocs.io/en/latest/
 
-.. primerend
+Additionally, our core geoprocessing routines have been open sourced as a separate python package, ``pygeoprocessing``, available here: https://github.com/natcap/pygeoprocessing.
```
I think James might have recently done some API docs for Pygeoprocessing?

emlys replied:
good point! Added the link to that

dcdenu4 commented on source/sdr.rst:
```diff
@@ -465,11 +460,10 @@ A key component for all water models is a spatially continuous land use/land cov
 
- *  NASA: https://lpdaac.usgs.gov/dataset_discovery/modis/modis_products_table/mcd12q1 (MODIS multi-year global landcover data provided in several classifications)
- *  The European Space Agency: https://www.esa-landcover-cci.org (Three global maps for the 2000, 2005 and 2010 epochs)
- *  The University of Maryland's Global Land Cover Facility: http://glcf.umd.edu/data/landcover/ (data available in 1 degree, 8km and 1km resolutions).
```
Is this deprecated?

dcdenu4 commented on source/wave_energy.rst:
```diff
@@ -73,19 +70,19 @@ Potential Wave Power Resource Assessment
 where the wave number :math:`k` is calculated using a dispersion relationship expressed as a function of wave frequency (:math:`w = 2\pi / T_e`) and water depth :math:`h`:
 
 .. math:: w^2 = {gk * \tanh(kh)}
-   :label: eq3
```
Again wondering if these are referenced in the source code.

emlys replied:
Same as above, this label shouldn't affect how it's labeled in the built docs.

dcdenu4 commented on source/wind_energy.rst:
```diff
@@ -70,21 +68,20 @@ where, :math:`k` and :math:`\lambda` are the shape and scale factors, respective
+Wind power density is calculated at the hub height :math:`Z` (m) of a wind turbine (:numref:`weibull-fig`), which means all variables in :eq:`wind_power_density` and :eq:`weibull_dist` need to be converted into the appropriate value at hub height. Mean air density :math:`\rho` was estimated as :math:`\rho=1.225-(1.194\cdot 10^{-4})Z`, which approximates the U.S. Standard Atmosphere profile for ai ...
 
 .. math:: \frac{V}{V_r} = \left(\frac{Z}{Z_r}\right)^\alpha
-   :label: eq3
```
I also wonder if keeping the equation number labels is more conforming to scientific papers as well. Might be more the "norm"

dcdenu4 reviewed (commented):
Hey @emlys , thanks for taking this on. Clearly needed a clean up. I left a few scattered comments about. It also looks like there are some merge conflicts currently.

emlys commented:
Thanks for the review @dcdenu4 ! Addressed your comments above.

emlys commented:
Sorry @dcdenu4 I added some more changes! They're all in a7331cf. Updating some deprecated options and removed the unused config template stuff (a lot of it is now deprecated). Also took out the makefile commands that we don't generally use, and the latex config options (was causing a bunch of warnings, and I don't think it's worth maintaining if we're not using latex).

dcdenu4 reviewed (commented):
Hey @emlys , this looks pretty good to me. I did notice that in the archived html version there is now an `index` button in the top right that sits by the `next` button. Clicking the `index` button sends me to a blank page though.

I was also wondering if anything needed to be updated in the InVEST repo makefile? I briefly looked in there and there's mention of creating a User Guide PDF. I'm not sure that actually happens. But just wanted to make sure there wasn't any coupled things we weren't thinking of.

Thanks!

emlys commented:
> I did notice that in the archived html version there is now an `index` button in the top right that sits by the `next` button. Clicking the `index` button sends me to a blank page though.

Haha I would have never noticed that! Fixed in 1773b9d

> I was also wondering if anything needed to be updated in the InVEST repo makefile? I briefly looked in there and there's mention of creating a User Guide PDF. I'm not sure that actually happens. But just wanted to make sure there wasn't any coupled things we weren't thinking of.

Good point, there isn't a PDF. I took out that comment and tidied up the `make userguide` command over in the apidocs PR: https://github.com/natcap/invest/pull/395/commits/4366ec2e475b60b5b6bf6a85b1eaf5de2c97a1ca

dcdenu4 reviewed (approved):
Great! Approved.
